### ISSUE TITLE
Reporting (link or imap) not working

### ISSUE THREAD

**Issue by quintingsnj** [gophish/gophish#2161]
**Thanks for reaching out! We're happy to help resolve issues as quickly as possible.**

**Please use this template when creating a new issue. If you do not follow this template format, your issue may be closed without comment.**

Before filing a new issue, please use the search bar at the top of the browser to search for similar issues. Also, please make sure you have read our [documentation](https://getgophish.com/documentation/), which covers how to use Gophish.

What version of Gophish are you using?: **0.11.0**

Brief description of the issue: **Reporting not working (either link, or imap)**

What are you expecting to see happen? : **I expect when clicking on the report link that it would mark it as reported, and/or when checking my imap account for reported spam, it would mark the corresponding record as reported.**

What are you seeing happen? **When clicking the link, nothing happens, for imap, I can see the email get marked as read when gophish checks it, but it does not mark the record as reported. I do note, that our spam reporting system requires the email to be sent as an attachment, so is gophish maybe not looking at attached emails?**

If this question is related to email templates or landing pages not working as expected, please provide your template or landing page below:

I know that issue #1985 talks about the link not working, but I downloaded and compiled the latest master (first time I have done that - so maybe I did something wrong)...anyways, replaced my gophish.exe with the newly compiled one, and no change.

Thanks!

**Comment by glennzw** [comment]
Hi there,

Would you be able to supply a sample email that's causing this behaviour so I can try reproduce it on my side? You can email it to me if you'd rather not post it publicly.

Also, when you say the 'reporting link is not working', do you mean when you push the report button on the gophish ui? Perhaps check the browser console too; if you're serving the admin page off SSL and the phishing page off plaintext you might see an error like this:

![Screenshot 2021-04-06 at 17 09 36](https://user-images.githubusercontent.com/3966613/113742862-f321c880-96fa-11eb-995f-57dbad559499.png)

The solution here would be to either server both off the same protocol, or allow mixed content in your browser (with caution).

**Comment by quintingsnj** [comment]
OK, for the link, that was EXACTLY the issue ☺

Thanks!

For imap/email, here is the .msg message that gets sent from our spam reporting system. Let me know if that let’s you see what you need.

Thanks!

Q

**Comment by glennzw** [comment]
Sorry, rather I meant could you share a campaign email that you're having trouble with? The .msg you provided seems to be our conversation here :)

**Comment by quintingsnj** [comment]
That’s weird….I checked, and the attachment I sent is exactly from our spam reporting system, with the campaign email attached to that…strange…I will attach just the campaign email… So, if an email gets sent to our imap box that has the campaign email attached to it, can gophish read that attached email as the campaign email?

Thanks,

Q

**Comment by quintingsnj** [comment]
Here is just the .msg for the email campaign…however, when reported, it is actually an attachment within another email…

Thanks.

Q

**Comment by quintingsnj** [comment]
I just realized that my ".msg" file attachment is getting stripped somewhere...so, here it is, linked, as a zipped file....

https://fileshare.snj.ca/download.php?id=963&token=C0NlwwVgjMDyikQFeWx6V4WMTEKy0yIG

Let me know if you can get it from there.

Thanks!

**Comment by glennzw** [comment]
Is the gophish email being to sent to your IMAP box as a .msg attachment? I'm not familiar with this 'CDFV2 Microsoft Outlook Message' format that it appears to be in (and neither is gophish). I can't actually find a way to open the file on my mac.

**Comment by quintingsnj** [comment]
It is being sent as a .msg attachment, as that is what our spam filter vendor requires (I think – at least this works for them)…If I need to convert it to a .eml format, I can look at my script that does the reporting, and see if I can convert it…


Q

**Comment by glennzw** [comment]
Yeah it'd need to be a .eml format, or inline. Basically gophish will just scan the email and eml attachments for the `?rid=XXXXX` parameter, indicating it's a campaign email


https://github.com/gophish/gophish/blob/master/imap/monitor.go#L224

**Comment by quintingsnj** [comment]
OK, sounds good…already found a command line converter, so hopefully can build that into my reporting script ☺

Thanks so much for your help, I think this gets me on the right track.

Q

**Comment by glennzw** [comment]
Good luck, and you're most welcome! Let us know how you get on.

**Comment by quintingsnj** [comment]
Hi Glenn. So I got an email with the campaign email attached in .EML format going to my imap account, but it doesn’t seem to work…I attached the .EML file to this email….

Q

**Comment by glennzw** [comment]
Hi Quintin - seems the attachment isn't showing up here. Could you email it to me? glennzw AT protonmail DOT com.

**Comment by quintingsnj** [comment]
Glenn, thanks so much for your help with this. I got this working now! I just made my script send it as an attachment for my spam vendor, and then send the actually phish email from the test as a regular email, and that is working!

Thanks again! Much appreciated.

Q

**Comment by glennzw** [comment]
Ah great news, nicely done Quintin! I'll keep working on a fix in case other people bump into it; basically the problem is that if an attached .eml is in quoted-printable format it's not being decoded, so we miss the rid parameter.

**Comment by blacklist-arcc** [comment]
Maybe the "Mark as reported" should also be done through an api request instead using only the phishing url. If the url is not valid anymore you can not mark as reported. Maybe we will write an api function for this and open a PR.
